app/src/main/java/cm/aptoide/pt/home/BundlesResponseMapper.java

```diff
@@ -66,14 +67,18 @@ public BundlesResponseMapper(InstallManager installManager,
         if (type.equals(HomeBundle.BundleType.APPS)
             || type.equals(HomeBundle.BundleType.EDITORS)
             || type.equals(HomeBundle.BundleType.TOP)) {
+
+          String widgetActionTag = getWidgetActionTag(widget);
           appBundles.add(new AppBundle(title, map(((ListApps) viewObject).getDataList()
-              .getList(), type, widgetTag), type, event, widgetTag));
+              .getList(), type, widgetTag), type, event, widgetTag, widgetActionTag));
         } else if (type.equals(HomeBundle.BundleType.APPCOINS_ADS)) {
+          String widgetActionTag = getWidgetActionTag(widget);
```

Comment:
Looks like you can just use one `String widgetActionTag = getWidgetActionTag(widget);` before all the if's instead of two.